portability-types-common/src/main/java/org/datatransferproject/types/common/models/music/MusicPlaylistItem.java

```diff
@@ -0,0 +1,104 @@
+/*
+ * Copyright 2022 The Data Transfer Project Authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.datatransferproject.types.common.models.music;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import com.fasterxml.jackson.annotation.JsonCreator;
+import com.fasterxml.jackson.annotation.JsonProperty;
+import com.google.common.base.MoreObjects;
+import com.google.common.base.Preconditions;
+import java.util.Objects;
+
+/**
+ * POJO for MusicPlaylistItem
+ */
+public class MusicPlaylistItem {
+
+  /**
+   * The MusicRecording belongs to the playlist item. Required.
+   */
+  private final MusicRecording track;
+
+  /**
+   * The playlist id of playlist item. Required. This ID should be unique within the upstream system
+   * from which the data originated. No particular format is guaranteed (eg: it might like a
+   * URI "music.acme.co/playlists/23d89fea", a plain ID "28921", or anything else really).
+   */
+  private final String playlistId;
+
+  /**
+   * The index of track inside the playlist. Best efforts to make sure the correct order of playlist
+   * items. Optional. Any int is valid - these numbers are meant purely to be relative to other
+   * MusicPlaylistItem {@code order} values
+   */
+  private final int order;
+
+  @JsonCreator
+  public MusicPlaylistItem(
+      @JsonProperty("track") MusicRecording track,
+      @JsonProperty("playlistId") String playlistId,
+      @JsonProperty("order") int order) {
+    Preconditions.checkNotNull(track, "track must be set for MusicPlaylistItem");
+    this.track = track;
+    if (isNullOrEmpty(playlistId)) {
+      throw new IllegalArgumentException("playlistId must be set for MusicPlaylistItem");
+    }
```

Comment:
nit: unify how we're checking state here

having two different approaches so close to eachother makes it look like you're purposely trying to get subtly different things, which I think is probably not what you want? eg: maybe unify them and pick `checkArgument` (since that's meant for this exact use case):

```java
    Preconditions.checkArgument(!isNullOrEmpty(track), "track must be set for MusicPlaylistItem");
    this.track = track;
    Preconditions.checkArgument(!isNullOrEmpty(playlistId), "playlistId must be set for MusicPlaylistItem");
    this.playlistId = playlistId;
```

Reply:
Updated in commit https://github.com/google/data-transfer-project/pull/1131/commits/96bcb7bd984c4c9cc45dc85616d3b0bbe0baf1a4.